File: puzzles/easy/ascii-art/ascii-art.py
# the standard input according to the problem statement.

l = int(input())
h = int(input())
t = input()
for i in range(h):
    row = input()
    out = ""
    for ch in t:
        ch_upper = ord(ch.upper())
        if 65 <= ch_upper <= 91:
            loc = ch_upper - 65
        else:
            loc = 26
        out += row[l * loc: l * (loc + 1)]
    print(out)

File: puzzles/easy/ascii-art/ascii-art.js
//the standard input according to the problem statement.

const L = parseInt(readline());
const H = parseInt(readline());
const T = readline();

for (let i = 0; i < H; i++) {
    const ROW = readline();
    let outLog = "";
    for (let i = 0; i < T.length; i++) {
        ch = T[i].toUpperCase().charCodeAt(0)
        let index = 26
        if (ch >= 65 && ch <= 91) {
            index = ch - 65
        }
        outLog += ROW.slice(index * L, (index + 1) * L);
    }
    console.log(outLog);
}
